cpp/include/cudf/strings/repeat_strings.hpp

```diff
@@ -78,13 +86,62 @@ std::unique_ptr<string_scalar> repeat_strings(
  * @param input The column containing strings to repeat.
  * @param repeat_times The number of times each input string is copied to the output.
  * @param mr Device memory resource used to allocate the returned strings column.
- * @return New column with concatenated results.
+ * @return New column with repeated strings.
  */
 std::unique_ptr<column> repeat_strings(
   strings_column_view const& input,
   size_type repeat_times,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Repeat each string in the given strings column by the number of times given in another
+ * numeric column.
+ *
+ * For a given strings column, an output strings column is generated by repeating each of its string
+ * by a number of times given by the corresponding row in a @p `repeat_times` numeric column.
+ *
+ * In the special cases:
+ *  - If any value in the `repeat_times` column is not a positive number and its corresponding input
+ *    string is not null, the output string will always be an empty string.
+ *  - Any null row (from either the input strings column or the `repeat_times` column) will always
+ *    result in a null output string.
+ *
+ * Using this API needs to follow a two-pass approach:
```

Comment:
@davidwendt Can you have a look at the new interface for this function again, please? I want to make sure this is what we want to have before actually implementing it.

Reply:
I found it a little strange returning two different column types for the same API but we do that for other APIs. Technically the behavior is different too though. Why create a `repeat_strings_output_sizes` separate API? Though at this point I feel this function could be created outside of libcudf since it is a simply dot-product of the input string lengths and the `repeat_times` vector. Anyway, this function could return the output_sizes column.

Then, the `repeat_strings` could take ths column as a `std::optional<column_view>` parameter. This way, the caller is not required to call the first API at all. If the sizes are passed, they are used to build the offsets. If not, the offsets are calculated like normal.

Reply:
I'm implementing this, leaving `compute_strings_output_sizes` completely to the caller as this can be achieved easily. For the `repeat_strings` API, I'm going to add `std::optional<column_view> output_strings_offsets` (offsets instead of string sizes), because the caller will have to compute strings offsets to check for overflow anyway.